### PR TITLE
Add DownloadFileFromRepo API to GitHub and GitLab

### PR DESCRIPTION
**Pull request by omerzi**
- [x] All [tests](https://github.com/jfrog/froggit-go/actions/workflows/test.yml) passed. If this feature is not already covered by the tests, I added new tests.
- [x] I used `go fmt ./...` for formatting the code before submitting the pull request.
- [x] This feature is included on all supported VCS providers - GitHub, Bitbucket cloud, Bitbucket server, and GitLab.

---

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -516,6 +517,26 @@ scanResults := "results"
 sarifID, err := client.UploadCodeScanning(ctx, owner, repo, branch, scanResults)
 ```
 
+#### Download File From Repository
```
Comment (yahavi) [inline README.md:520]:
```suggestion
#### Download a File From a Repository
```

File: vcsclient/github.go [hunk 2]
```diff
@@ -503,6 +503,27 @@ func (client *GitHubClient) UploadCodeScanning(ctx context.Context, owner, repos
...
+		return nil, 0, err
+	}
+	body, response, err := ghClient.Repositories.DownloadContents(ctx, owner, repository, path, &github.RepositoryContentGetOptions{Ref: branch})
+	if err != nil {
+		return nil, response.StatusCode, err
+	}
+	if response.StatusCode != http.StatusOK {
+		return nil, response.StatusCode, fmt.Errorf("expected %d status code while received %d status code", http.StatusOK, response.StatusCode)
+	}
+
```
Comment (yahavi) [inline vcsclient/github.go:519]: Should we close the body here? 
https://github.com/google/go-github/blob/v49.0.0/github/repos_contents.go#L122

File: vcsclient/github_test.go [hunk 3]
```diff
@@ -210,6 +210,23 @@ func TestGitHubClient_DownloadRepository(t *testing.T) {
...
 
+func TestGitHubClient_DownloadFileFromRepository(t *testing.T) {
+	ctx := context.Background()
+	downloadURL := "https://jfrog.com"
+	name := "hello-world"
+	client, cleanUp := createServerAndClient(t, vcsutils.GitHub, false, &[]github.RepositoryContent{{DownloadURL: &downloadURL, Name: &name}}, "/repos/jfrog/repo-1/contents/?ref=branch-1", createGitHubHandler)
+	defer cleanUp()
+	_, statusCode, err := client.DownloadFileFromRepo(ctx, owner, repo1, branch1, "hello-world")
+	assert.NoError(t, err)
+	assert.Equal(t, http.StatusOK, statusCode)
```
Comment (yahavi) [inline vcsclient/github_test.go:221]: We can also check that the response is not empty
```suggestion
	content, statusCode, err := client.DownloadFileFromRepo(ctx, owner, repo1, branch1, "hello-world")
	assert.NoError(t, err)
	assert.Equal(t, http.StatusOK, statusCode)
	assert.NotEmpty(t, content)
```

**Review by yahavi — CHANGES_REQUESTED** [review]

File: README.md (hunk 1 above)
Comment (omerzi) [reply README.md:520]: done

File: vcsclient/github.go (hunk 2 above)
Comment (omerzi) [reply vcsclient/github.go:519]: done

File: vcsclient/github_test.go (hunk 3 above)
Comment (omerzi) [reply vcsclient/github_test.go:221]: done

**Pull request merged**